base/abstract_interface.hpp

```diff
@@ -6,32 +6,30 @@
 namespace MRA
 {
 
-template <typename Ti, typename Tp, typename Ts, typename To, typename Tl>
+template <typename T_InputType, typename T_ParamsType, typename T_StateType, typename T_OutputType, typename T_LocalType>
```

Comment:
T_...Type is a bit redundant. Why not just InputType. Actually, why did you change this from Ti etc?

Reply:
because InputType is defined in the class ca 8 lines below this line.